### PR TITLE
Feature/interface

### PR DESCRIPTION
**Pull request by epinczinger**
Submitting project for code review. This app was designed following the given instructions.

- A user can enter a city and search for its weather conditions.
- Error handling was applied.
- A descriptive readme was added.
- Basic styling added.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "eslint-plugin-import": "^2.20.2",
+    "scss-loader": "0.0.1",
+    "style-loader": "^2.0.0",
+    "stylelint": "^13.3.3",
+    "stylelint-config-standard": "^20.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.17.2",
+    "vue-style-loader": "^4.1.2",
+    "webpack": "^5.9.0",
+    "webpack-cli": "^4.2.0"
```
Comment (addod19) [inline package.json]: Kindly add `webpack dev server` so we can run the server successfully. Currently, `npm run start` returns error

File: src/components/api.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const infoAPI = async (city, units = 'metric') => {
+  const apiKey = 'f8584d3f19d9882aec61e1e892543fe7';
+  const response = await fetch(
+    `http://api.openweathermap.org/data/2.5/weather?q=${city}&units=${units}&appid=${apiKey}`,
+  );
+  if (response.status === 200) {
+    const info = await response.json();
```
Comment (addod19) [inline src/components/api.js:7]: Kindly add a `try {}` and `catch {}` block

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack",
```
Comment (addod19) [inline package.json]: Kindly change `"build": "webpack"` to `"build": "webpack --mode production"`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "watch-build": "webpack --watch",
+    "dev": "webpack --mode development",
+    "start": "webpack serve --mode development --open 'google-chrome'"
```
Comment (addod19) [inline package.json:30]: [OPTIONAL]
- Kindly change the start script to ``"start": "webpack-dev-server --mode development --open"``

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @epinczinger, great job on this project :+1: Great effort working on this project :smile: However, there are a couple of issues we need to address. Kindly check the comments below:
- The temperature toggle is currently not working as expected. It only works when the data is submitted and fails to work when selected. Let us improve it
- The appearance must change for every data received. Currently there appearance does not change. Let us fix it

[Optional]
- Kindly implement a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API
- KIndly implement fetch request to retrieve the current geolocation of the user when the app is first accessed and fetches the weather from that location

Once these are fixed, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: package.json (hunk 1 above)
Comment (epinczinger) [reply package.json]: Done

File: package.json (hunk 3 above)
Comment (epinczinger) [reply package.json]: Done

File: package.json (hunk 4 above)
Comment (epinczinger) [reply package.json:30]: In a computer with Linux Mint, this is the correct way to run it.

File: package.json (hunk 1 above)
Comment (epinczinger) [reply package.json]: Done

File: src/components/api.js (hunk 2 above)
Comment (epinczinger) [reply src/components/api.js:7]: Done

**Comment by epinczinger** [comment]
Thanks for the review @addod19. Required changes have been done. 
Requesting for another code review.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi @epinczinger  👋

**You've done pretty well here.💪**

However, let's still fix a few issues and we are good to go.

- [x] Please **after the user receives the weather data** make it possible for the user to toggle temperature.

- [x] Also you are expected to change at least the background of the page for different weather types. If a location is cold it should show a different background than when it is hot. Please correct this.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by epinczinger** [comment]
HI @mikenath223, thanks for the review. The background of the page was already changing when submitting the city, please feel free to download and review the code again, maybe it was the live link that was not working properly. It was fun to add the toggle button, It was not on the requirement though. Submitting again for another code review.

Regards.

**Comment by mikenath223** [comment]
> HI @mikenath223, thanks for the review. The background of the page was already changing when submitting the city, please feel free to download and review the code again, maybe it was the live link that was not working properly. It was fun to add the toggle button, It was not on the requirement though. Submitting again for another code review.
> 
> Regards.

Rest assured that your code was downloaded and run locally @epinczinger 🙂

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Comment by epinczinger** [comment]
Thanks Mike!!

**Pull request merged**